### PR TITLE
GradeSummaries and HWReport page ports

### PR DESCRIPTION
**Pull request by jbarthelmess**
- Pages should create GradeSummaries and HWReports correctly
- Currently nothing visible happens when the buttons are pushed.  I was hoping to add a green message telling the user it was successful, I'm open to other suggestions. 
- Also not sure what the idle "in progress" indication should be, maybe just add a simple wheel gif in the button or something.
- Note the javascript was an attempt to add a green message, doesn't work.  The AJAX works but the resulting frontend doesn't.

### CODE REVIEW

**Comment by jbarthelmess** [comment]
I don't use those two functions anywhere, but the TA Grading page needs them to make the Student information panel, so I included them for James

File: site/app/models/LateDaysCalculation.php [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+        return $all_latedays;
+    }
+    
+     /**
+     * For the given user id generate the late day usage HTML table.
+     * @param $user_id String. The user id of the user whose table you want.
+     * @return string. The string representation of the HTML table.
+     */
+    public function generate_table_for_user($user_id){
```
Comment (jbarthelmess) [inline site/app/models/LateDaysCalculation.php]: I don't use this function or the one below it, but they are used to create the student info panel in the TA Grading page, so I added them to be used when James needs that info for the TA Grading page port. We will probably want to move those functions to the TA Grading view when we get rid of this class and move it's info into the Gradeable model

**Comment by jbarthelmess** [comment]
I removed them, but just out of curiosity, why do we omit the closing tag ?> ?

**Comment by MasterOdin** [comment]
Primarily, for [PSR-1](http://www.php-fig.org/psr/psr-1/) which is the main coding style guide that governs the code base (with some custom tweaks, outlined http://submitty.org/developer/coding_style_guide).

However, the reason for that is (from the [PHP docs](http://php.net/manual/en/language.basic-syntax.phptags.php)):

> If a file is pure PHP code, it is preferable to omit the PHP closing tag at the end of the file. This prevents accidental whitespace or new lines being added after the PHP closing tag, which may cause unwanted effects because PHP will start output buffering when there is no intention from the programmer to send any output at that point in the script.

So it helps to enforce the rule that all .php files are pure PHP code (TAGradingSite used to not be) and for the above reason.

**Pull request merged**